pkg/client/client.go

```diff
@@ -545,9 +545,19 @@ func (c *Client) CreateOrUpdateDeployment(dep *appsv1.Deployment) error {
 		// deployment is equivalent to the one that would be applied.
 		return nil
 	}
-
 	err = c.UpdateDeployment(dep)
-	return errors.Wrap(err, "updating deployment object failed")
+	if err != nil {
+		// try to delete deployment
+		err = c.DeleteDeployment(dep)
```

Comment:
only do this if the error is a conflict, see here for an example https://github.com/coreos/prometheus-operator/blob/31bf4b889a5ea6de4c1907e299066787fcd73bae/pkg/prometheus/operator.go#L1178

Reply:
PTAL @brancz thanks

Reply:
lgtm